src/core/instance/render-helpers/bind-object-props.js

```diff
@@ -1,8 +1,11 @@
 /* @flow */
 
 import config from 'core/config'
+import { makeMap } from 'shared/util'
 import { isObject, warn, toObject } from 'core/util/index'
 
+const isSpecialAttributes = makeMap('key,ref,slot,is')
```

Comment:
There is a similar util [here](https://github.com/vuejs/vue/blob/dev/src/core/instance/state.js#L57-L61)

Let's extract it into `shared/util` as `isReservedAttribute` to avoid duplication.